nix/cudaPackagesExtensions/0001-add-latest-nccl.nix

```diff
@@ -0,0 +1,13 @@
+{ fetchFromGitHub }:
+ffinal: pprev: {
+  nccl_latest = pprev.nccl.overrideAttrs (prevAttrs: {
+    src = fetchFromGitHub {
+      owner = "NVIDIA";
+      repo = "nccl";
+      rev = "v2.23.4-1";
+      hash = "sha256-DlMxlLO2F079fBkhORNPVN/ASYiVIRfLJw7bDoiClHw=";
+    };
+    name = "cuda${ffinal.cudaMajorMinorPatchVersion}-nccl-2.23.4-1";
```

Comment:
Is it necessary to change the name? Using `backendStdenv`, `name` should be the result of `${backendStdenv.cudaNamePrefix}-${pname}-${version}` (https://github.com/ConnorBaker/cuda-packages/blob/bb6bf09b919e8e7dda006bcee9a48675d034850f/cuda-packages/common/backendStdenv.nix#L55) by default, so updating the version should be enough.